### ISSUE TITLE
OSK is not shown sometimes

### ISSUE THREAD

**Issue by lduboeuf** [ubports/ubuntu-touch#1268]
### Illustrations

- Device: N5
- Channel: Edge
- Build: Latest

### Steps to reproduce

While i want to reply to a sms or use keyboard to input text, the OSK is not displayed (blank area, or nothing displayed )

### Expected behavior

OSK is displayed and usable

### Actual behavior

Randomly, i have no keyboard displayed  It occurs on different apps ( Morph Browser, Contact app, SMS )

### Logfiles and additional information

![screenshot20191023_224118150](https://user-images.githubusercontent.com/11663835/67459832-bdd60780-f639-11e9-88e4-9a680197c82c.png)

**Comment by UniversalSuperBox** [comment]
The issues "the keyboard rectangle does not appear" and "the keyboard rectangle is empty" are separate, and I'm going to take over this report only for "the keyboard rectangle is empty".

I've seen this issue many times, but only tried to force the keyboard to show the problem today. It appears only once in a great many opens of the keyboard, but there's an interesting difference in the Unity8 log between a couple working and a non-working show:

```[2019-10-30:20:47:27.286] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(restored)
[2019-10-30:20:47:27.288] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateExposure(true)
[2019-10-30:20:47:28.325] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(hidden)
[2019-10-30:20:47:28.326] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateExposure(false)
[2019-10-30:20:47:28.910] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(restored)
[2019-10-30:20:47:28.911] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateExposure(true)
[2019-10-30:20:47:29.776] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(hidden)
[2019-10-30:20:47:29.777] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateExposure(false)

[2019-10-30:20:47:30.321] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(restored)
[2019-10-30:20:47:31.168] qtmir.surfaces: MirSurface[0x3135118,"maliit-server"]::updateState(hidden)```

We're getting the `updateState` call but not the `updateExposure` call when that happens. Quite interesting!

**Comment by james-lyon** [comment]
I agree that there are two separate issues. When I encounter this issue (keyboard rectangle is empty) on my Aquaris E4.5, I can still type characters if I tap the screen in the right place, and the OSK displays very briefly when I do so. So for me the OSK is there but invisible.

**Comment by Danfro** [comment]
I can confirm that this happens sometimes/rarely on my E5 too. And it is the same as described above, I can still type characters but can't see them.

**Comment by UniversalSuperBox** [comment]
@Danfro and @james-lyon, indeed, these are two separate issues. I take this one to mean "blank background, no keys are available and the rectangle is not interactive," so if you aren't able to find an issue filed for what you're experiencing please file a new one.

**Comment by UniversalSuperBox** [comment]
Closed by https://github.com/ubports/qtmir/pull/41

**Comment by lduboeuf** [comment]
congrats for all this work

**Comment by giiba** [comment]
I'm not sure if this is related, or how to reproduce it. But I turned on the screen of my device yesterday to find:
![screenshot20200216_121709552](https://user-images.githubusercontent.com/7121141/74692997-9bdf3280-519e-11ea-9b9e-d39a79a983d2.png)

I had to swipe down to dismiss the keyboard before I could swipe to unlock the screen. I am sorry to say that try as I might I cannot reproduce it.

I am on Development channel updates, on a OPO.

**Comment by UniversalSuperBox** [comment]
Hi @giiba,

That does not sound like the same issue listed in this report. Please submit a new issue. Thanks!

**Comment by jonnius** [comment]
Have not seen this issue any more on dev recently and also not on RC now on my FP2. So I confirm this fixed.

**Comment by cibersheep** [comment]
I use Arale as daily drive (Dev) and keyboard is reliable

**Comment by lduboeuf** [comment]
+1
